### PR TITLE
Add recipe for skerrick

### PR DESCRIPTION
**Pull request by anonimitoraf**
### Brief summary of what the package does

REPL-driven development for Javascript! See https://github.com/anonimitoraf/skerrick#why-should-i-care for more info

### Direct link to the package repository

https://github.com/anonimitoraf/skerrick

### Your association with the package

Author

### Relevant communications with the upstream package maintainer

None needed

### Checklist

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

### CODE REVIEW

File: recipes/skerrick [hunk 1]
```diff
@@ -0,0 +1,3 @@
+(skerrick :repo "anonimitoraf/skerrick"
+          :fetcher github
+          :files ("skerrick.el"))
```
Comment (riscy) [inline recipes/skerrick:3]: Equivalent but gives your package room to grow:
```suggestion
(skerrick :repo "anonimitoraf/skerrick"
          :fetcher github)
```

**Review by riscy — COMMENTED** [review]
Thanks for this!  Very quick first pass --

`skerrick.el` with byte-compile using Emacs 27.1:
```
skerrick.el:156:1:Warning: the function `length>' is not known to be defined.
```
I couldn't find this function anywhere -- maybe you are running a later version of Emacs?

`skerrick.el` with [melpazoid](https://github.com/riscy/melpazoid):
- skerrick.el#L104: Loading a package should rarely add hooks.  Especially a global hook to `post-command-hooks` -- can this be gated behind a mode?  Can it be restricted to local to the buffer only?

Package and license:
- The summary line and Commentary are incomplete; the summary line for example just reads "Description"
- Add a standard license header or an [SPDX-License-Identifier](https://spdx.dev/ids/) to skerrick.el

**Comment by anonimitoraf** [comment]
Hey @riscy, thanks for the quick reply!

Ah, slipped my mind that I'm on Emacs 29. I'll set up a minimal Emacs environment and try compile there.

I'll gate the hook inside a minor mode. Would it also make sense to gate the function definitions (i.e. the `defun`s) inside the mode too?

I'll fix up the package commentary and license

Thanks!

**Comment by riscy** [comment]
> Would it also make sense to gate the function definitions (i.e. the defuns) inside the mode too?

Thanks for accommodating!  The defuns should stay at the top level.  One way that might be helpful to think of it is to make your package as much like an importable library as possible.  A well-behaved library shouldn't be doing much work or changing much of the runtime on import.  A somewhat related line of thinking: https://instagram-engineering.com/python-at-scale-strict-modules-c0bb9245c834

**Comment by anonimitoraf** [comment]
Algood! I see. But the hook in this case is somewhat related to one of the functions. I'm envisioning a scenario where:

* The lib gets imported BUT the minor mode does not get invoked (so hook doesn't get registered)
* The top-level functions are invoked, but because the mode/hooks aren't invoked, the overlays linger on the screen (which is bad UX)

**Comment by anonimitoraf** [comment]
Actually, perhaps, I should put the hooks inside the relevant function/s?

**Comment by anonimitoraf** [comment]
I've made the changes btw

**Comment by riscy** [comment]
I got your note!  Just fyi I'm usually most active here on weekends.

**Comment by anonimitoraf** [comment]
Ah, algood. Thanks for letting me know :)

**Comment by riscy** [comment]
Thank you!  Had another look, and I think the summary is still incomplete (it says "Description").  Otherwise this is looking ready for a quick merge.

**Comment by anonimitoraf** [comment]
Hey, sorry, I didn't see the "Description" at the top whoops. Filled it in now

**Comment by riscy** [comment]
Looks good, thanks!  And welcome to MELPA. :)

**Comment by anonimitoraf** [comment]
Thanks!

**Pull request merged**